### PR TITLE
Bonding address

### PR DESCRIPTION
**Pull request by Enciusan**
### Description

Check all boxes that apply and review your code if you can't tick them all yet:

- [ ] Did you test your changes in logged out, mainnet logged in and devnet logged in modes?
- [ ] Did you test your changes in mobile, tablet and desktop modes?
- [ ] Did you test your changes with or without the Itheum Web2 API availability? (don't forget about "graceful fallback")
- [ ] Did you use good user alerts and error messages? (and put them in language.ts)
- [ ] Did you leave the code base cleaner than you found it overall (done reasonable refactoring efforts wherever you felt that was needed)?

### CODE REVIEW

**Comment by michavie** [comment]
@Enciusan we have released a Deep Link Builder yesterday to make it easy to generate deep links: 

NPM: https://www.npmjs.com/package/@peerme/sdk

Usage:

Screenshot 2024-04-12 at 12 32 21 PM

**Comment by Enciusan** [comment]
Hello @michavie i found out this error pops up when i try to use it :) 
image

**Comment by michavie** [comment]
@Enciusan Importing like this does not work?

```
import { ProposalDeepLinkBuilder } from '@peerme/sdk'
```

We may need to add a default export to use it similar to this:

```
import PeerMe from '@peerme/sdk'
PeerMe.ProposalDeepLinkBuilder
```

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "@multiversx/sdk-dapp": "2.28.7",
     "@multiversx/sdk-native-auth-client": "1.0.7",
     "@multiversx/sdk-network-providers": "2.3.0",
+    "@peerme/sdk": "^0.2.1",
```
Comment (damienen) [inline package.json:17]: This should have a stable version

File: src/pages/Bonding/Bonding.tsx [hunk 2]
```diff
@@ -24,11 +24,15 @@ export const Bonding: React.FC = () => {
   const navigate = useNavigate();
 
   const checkIfUserIsAdmin = () => {
+    const splittedAddresses = bondContractAdminDevnet.split(", ");
```
Comment (damienen) [inline src/pages/Bonding/Bonding.tsx:27]: Here it should be a bit less restrictive.

Just split by "," and then use trim() to get rid of white spaces

File: src/pages/Bonding/Bonding.tsx [hunk 3]
```diff
@@ -24,11 +24,15 @@ export const Bonding: React.FC = () => {
...
     if (!address) return false;
-    const adminAddress = IS_DEVNET ? bondContractAdminDevnet : bondContractAdminMainnet;
-    return address === adminAddress;
+    const adminAddress: any = IS_DEVNET ? splittedAddresses : bondContractAdminMainnet;
+    console.log(adminAddress.includes(address), address, adminAddress, address === adminAddress);
+    return adminAddress.includes(address) || address === adminAddress;
   };
 
+  // console.log()
```
Comment (damienen) [inline src/pages/Bonding/Bonding.tsx:34]: Delete this comment

File: src/pages/Bonding/Bonding.tsx [hunk 4]
```diff
@@ -24,11 +24,15 @@ export const Bonding: React.FC = () => {
   const navigate = useNavigate();
 
   const checkIfUserIsAdmin = () => {
+    const splittedAddresses = bondContractAdminDevnet.split(", ");
     if (!address) return false;
-    const adminAddress = IS_DEVNET ? bondContractAdminDevnet : bondContractAdminMainnet;
-    return address === adminAddress;
+    const adminAddress: any = IS_DEVNET ? splittedAddresses : bondContractAdminMainnet;
+    console.log(adminAddress.includes(address), address, adminAddress, address === adminAddress);
```
Comment (damienen) [inline src/pages/Bonding/Bonding.tsx:30]: Deleete this log

File: src/pages/Bonding/components/BondingParameters.tsx [hunk 5]
```diff
@@ -89,6 +96,48 @@ export const BondingParameters: React.FC = () => {
...
+    const amountOfTimeAfter = timeUntil(period);
+    const url = new ProposalDeepLinkBuilder("itheum-dao", { network: IS_DEVNET ? "devnet" : "mainnet" })
+      .authenticate(tokenLogin?.nativeAuthToken ?? "")
+      .setTitle("Set New Period Bond")
+      .setDescription(
+        `This is to propose a change in the Period Bond from ${amountOfTimeBefore.count} ${amountOfTimeBefore.unit} second and ${BigNumber(contractConfiguration.lockPeriodsWithBonds[0].amount).dividedBy(10 ** 18)} ITHEUM to  ${amountOfTimeAfter.count} ${amountOfTimeAfter.unit} and ${amount} ITHEUM `
+      )
+      .addAction(bondContract.getContractAddress().bech32(), "addPeriodsBonds", 0, [period, BigInt(amount) * BigInt("1000000000000000000")], [])
+      .build();
+    // console.log(url);
```
Comment (damienen) [inline src/pages/Bonding/components/BondingParameters.tsx:130]: Delete comment

File: src/pages/Bonding/components/BondingParameters.tsx [hunk 6]
```diff
@@ -89,6 +96,48 @@ export const BondingParameters: React.FC = () => {
...
+    setConstructedDeeplinkURL(url);
+    return window.open(url, "_blank");
+  };
+
+  // const earlyWithdrawalDeeplinkUrl = (newEarlyWithdrawalValue: number) => {
```
Comment (damienen) [inline src/pages/Bonding/components/BondingParameters.tsx:135]: Delete comments

**Review by damienen — CHANGES_REQUESTED** [review]

**Review by damienen — APPROVED** [review]

**Review by damienen — APPROVED** [review]

**Pull request merged**